src/main/scala/dxWDL/compiler/GenerateIRTask.scala

```diff
@@ -106,18 +107,51 @@ case class GenerateIRTask(
   def apply(task: CallableTaskDefinition, taskSourceCode: String): IR.Applet = {
     Utils.trace(verbose.on, s"Compiling task ${task.name}")
 
+    def unwrapParamMeta(
+        paramMeta: Option[MetaValueElement]
+    ): Option[Vector[IR.IOAttr]] = paramMeta match {
+      case None => None
+      case Some(MetaValueElementObject(obj)) => {
```

Comment:
I think you can reduce this to by matching the expected case early ( instead of having nested case/match) since they are all returning `None`.


```
   case Some(MetaValueElementObject(obj)) => {
        // Use flatmap to get the "help" and "pattern" keys if they exist
        Some(obj.flatMap {
            case ("help", MetaValueElementString(text)) => Some(IR.IOAttrHelp(text))
            case ("patterns", MetaValueElementArray(array)) =>
                Some(IR.IOAttrPatterns(array.flatMap {
                    case MetaValueElementString(pattern) => Some(pattern)
                    case _                               => None
                }.toVector))
            case _ => None
        }.toVector)
    }
```

Reply:
Excellent point!